### ISSUE TITLE
Feature: New carrier setup with m.carrier_add

### ISSUE THREAD

**Issue by cimspin** [PilotsTradeNetwork/MissionAlertBot#26]
m.carrier_add presently adds an FC to the internal database. However the information supplied could be used to automate the setup of Discord owner role ("P.T.N. NEWCARRIER (REG-NUM)"), crew role ("CREW: Newcarrier") and channel creation ("#ptn-newcarrier") which are all presently done manually. Channel would have to be created before ID is scraped for database but this would be helpful in preventing errors anyway.

If used in concert with issue #25 (crew roles command) would fully automate new carrier entry.

**Comment by cimspin** [comment]
The process I go through is this:
> 1. Create discord channel, it's just a # in front of whatever would be entered for discord-channel (but if the bot does it it has to be done before it tries to find the channel and save its ID to the db). This is basically the carrier's name without punctuation so should be able to derive from the name input, also swapping spaces for hyphens (-).
> 2. Create crew role in the format "CREW: {carrier name in title case without PTN prefix}" - though we can easily change this to be "CREW: {full carrier name in uppercase}" to make it easier for us.

Then I use Zira to create a crew signup but I'm hoping we can change this to be an m.crew command in a carrier's channel (similar to how m.ission and m.complete work). So if we implement auto setup we should probably do it that way and reform the crew signup system at the same time. Presently it's:
> ~~3a. Use Zira to create an embed in #crew-signups with standard title: "CREW: {full carrier name in title case}" - though again we can be flexible on this - and description: "Click :loudspeaker: to subscribe to #carrier-discord-channel trade alerts."~~
> ~~4a. Use Zira to add a reaction role to the above embed, format is z/normal :loudspeaker: @role~~
But it'd be better as:
> 3b. Create role as above, and have that role called by m.crew command when used in a carrier's channel (issue #25)

Then there's the bit I'm hoping to retire completely very soon, which is the carrier owner's title:
> 5. Create role in format "{full carrier name in uppercase} ({full carrier reg in uppercase})"
> 6. Assign to carrier's owner
> This last is janky because we only do it for main carriers. It's a way to associate owners with carriers so users can easily see but honestly it sucks, we've got way too many roles and there are better ways to do this.

So basically, only steps 1, 2 and 3b would really be necessary if it was automated
